tests/components/FilterSearch.test.tsx

```diff
@@ -659,7 +659,19 @@ describe('search without section labels', () => {
     await userEvent.type(screen.getByRole('textbox'), 'a');
     expect(mockedOnDropdownInputChange).toHaveBeenCalledTimes(1);
     expect(executeFilterSearch).toHaveBeenCalledTimes(0);
-  })
+  });
+
+  it('when an afterDropdownInputFocus prop is provided, invokes it in addition to the original ' +
+    'behavior when input gains focus', async () => {
+    const mockedAfterDropdownInputFocus = jest.fn();
+    const executeFilterSearch = jest.spyOn(SearchHeadless.prototype, 'executeFilterSearch');
+    renderFilterSearch(
+      {searchFields: searchFieldsProp, afterDropdownInputFocus: mockedAfterDropdownInputFocus});
+    await userEvent.click(screen.getByRole('textbox'));
+    expect(mockedAfterDropdownInputFocus).toHaveBeenCalledTimes(1);
+    await userEvent.type(screen.getByRole('textbox'), 'a');
```

Comment:
actually, I thought about it a little more, and this is technically still testing that filter search is executed on input change rather than focus. can we type a character in the input before we create the spies/mocks or reset them after typing a character? then we can check that clicking on the textbox calls both functions

Reply:
Updated!